src/ImageSharp/Common/Helpers/SimdUtils.HwIntrinsics.cs

```diff
@@ -221,7 +226,7 @@ public static class HwIntrinsics
                 ref Vector256<float> destBase =
                     ref Unsafe.As<float, Vector256<float>>(ref MemoryMarshal.GetReference(dest));
 
-                nint n = (nint)(uint)(dest.Length / Vector256<float>.Count);
+                nint n = (nint)((uint)dest.Length / (uint)Vector256<float>.Count);
```

Comment:
I've forgotten these in the other PR (I think because they're kept as `nint` due the operation in L231).

Reply:
I wonder if we should create a helper method for these? Feels very noisy seeing this before all our SIMD loops.

```C#
internal static partial class Numerics
{
    // Not sure about the name
    public static nint VectorCount<T>(Span<T> dest, int vectorSize) => (nint)((uint)dest.Length / (uint)vectorSize);
}
```

So then we can do:

```C#
nint n = Numerics.VectorCount(dest, Vector256<float>.Count);
```

Reply:
@JimBobSquarePants meh or yeah?

Reply:
I like the idea.
But it should be generic to account for different vector types. 

Maybe as extension to span?

Reply:
> But it should be generic to account for different vector types

I wish there was an `IIntrinsicVector<T>` interface with a static `Count` property but there isn't, and we are not on `net7.0+` anyways :) So I'm just passing `Vector256<float>.Count` as a parameter in my recommendation, meaning that there is no way to check if the `Vector256<T>` and `Span<T>` element types are the same. Not ideal, but still better than writing out the verbose `uint` magic everywhere IMHO.

Reply:
> no way to check if the `Vector256<T>` and `Span<T>` element types are the same

They don't need to match. Reading e.g. `Vector128<float>` out of a `ROS<byte>` is pretty common.

So for the generic variant there could be [two generic arguments](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAJI8ovLrl5hcAbho0AxFw4AbFdmAqYDACa9cGrfNoBmJqQYAxCBAY0A3jQbOmZ5eIwMAggAoAygAO2FwAPMAAnhgwAHwMuEFcAJQMALyx8cEsAGowYBjQAMIQ3BjkpAAcYZEwaAwAZioQ2BjRPoly1AC+xh4wUFzYKkzkSOYMALLhAKIIURK8EBIOTi7EZsyj7jwMOXmFxTxllQAqtce7+VCtGAAW+gyBwSHH6QmJKy4uAO43fdrnuUuDBAcQwUA4eQ+DEc1E+n2IAHYGD4OB5EhkuCwADIwLgAc1uDAA9MjUTxEhdoEdnpSriwiiUOp9ul0gA=). Here I don't like that the type of the span must be repeated. That should be infered (by the compiler).

In [your](https://github.com/SixLabors/ImageSharp/pull/2419#discussion_r1149799246) I don't like that `VectorXYZ<Type>.Count` needs to be typed.

What do you think of [this approach](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAJI8ovLrl5hcAbho0AxFw4AbFdmAqYDACa9cGrfNoBmJqQYAxCBAY0A3jQbOmZ5eIwMAggAoAygAO2FwAPMAAnhgwAHwMuEFcAJQMALyx8cEsAGowYBjQAMIQ3BjkpAAcIQBmKhDYGNE+iXLULq4M7jwMAEL+CWGRMXEJyWnDmTl5hcU8pACsSNW19Y3NNAC+xh4wUFzYKkzkSOYMALLhAKIIURK8EBIOTi7EZszHnZ6T+VBFJWWVABUvtBGhgABb6BiBYIDKLpEZPNrOADuYJ22iBuW+DBAcQwUA4eURSLGPg4HkSGS4LAAMjAuABzcEMAD0DDJFOBUH+IUxUyg0RYvx4LTaxJeh3e5K6XOFGHmiz531BENwUP6ETh4ySxLaqPRDCV0BxeIJRNaSOcpOlGEpCVp9KZYNZ7JtiS5Ct5XMFcpa6yAA==) (IIRC it's only ROS<byte>/Span<byte>)?

Reply:
> What do you think of [this approach](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAJI8ovLrl5hcAbho0AxFw4AbFdmAqYDACa9cGrfNoBmJqQYAxCBAY0A3jQbOmZ5eIwMAggAoAygAO2FwAPMAAnhgwAHwMuEFcAJQMALyx8cEsAGowYBjQAMIQ3BjkpAAcIQBmKhDYGNE+iXLULq4M7jwMAEL+CWGRMXEJyWnDmTl5hcU8pACsSNW19Y3NNAC+xh4wUFzYKkzkSOYMALLhAKIIURK8EBIOTi7EZszHnZ6T+VBFJWWVABUvtBGhgABb6BiBYIDKLpEZPNrOADuYJ22iBuW+DBAcQwUA4eURSLGPg4HkSGS4LAAMjAuABzcEMAD0DDJFOBUH+IUxUyg0RYvx4LTaxJeh3e5K6XOFGHmiz531BENwUP6ETh4ySxLaqPRDCV0BxeIJRNaSOcpOlGEpCVp9KZYNZ7JtiS5Ct5XMFcpa6yAA==) (IIRC it's only ROS<byte>/Span<byte>)?

I like this option the most. We can have one overload for each span type.

Reply:
Super.

I prefer doing this in a separete PR, as it doesn't match with this PR (this location is a left-over from the other PR) and there a quite a few places that need to be touched.
Fine?

Anyway I'll do this change later this afternoon (here or in a new PR).

Reply:
--> https://github.com/SixLabors/ImageSharp/pull/2422